integration/src/bitcoin/testnet.ts

```diff
@@ -44,94 +31,100 @@ export function testnetTests(get: () => { wallet: HDWallet; info: HDWalletInfo }
-            scriptType: BTCOutputScriptType.PayToP2SHWitness,
-            addressType: BTCOutputAddressType.Change,
+            addressNList: core.bip32ToAddressNList("m/49'/1'/0'/1/0"),
+            scriptType: core.BTCOutputScriptType.PayToP2SHWitness,
+            addressType: core.BTCOutputAddressType.Change,
             amount: String(123456789 - 11000 - 12300000),
             isChange: true,
           },
         ];
-        let res = await wallet.btcSignTx({
+        const res = await wallet.btcSignTx({
           coin: "Testnet",
-          inputs: inputs,
-          outputs: outputs,
+          inputs: inputs as core.BTCSignTxInput[],
+          outputs,
           version: 1,
           locktime: 0,
         });
-        expect(res.serializedTx).toEqual(
+        expect(res?.serializedTx).toEqual(
           "0100000000010137c361fb8f2d9056ba8c98c5611930fcb48cacfdd0fe2e0449d83eea982f91200000000017160014d16b8c0680c61fc6ed2e407455715055e41052f5ffffffff02e0aebb00000000001976a91414fdede0ddc3be652a0ce1afbc1b509a55b6b94888ac3df39f060000000017a91458b53ea7f832e8f096e896b8713a8c6df0e892ca8702483045022100ccd253bfdf8a5593cd7b6701370c531199f0f05a418cd547dfc7da3f21515f0f02203fa08a0753688871c220648f9edadbdb98af42e5d8269364a326572cf703895b012103e7bfe10708f715e8538c92d46ca50db6f657bbc455b7494e6a0303ccdb868b7900000000"
         );
       },
       TIMEOUT
     );
 
     test("btcSignTx() - p2wpkh", async () => {
-      if (!wallet || isPortis(wallet)) return;
-      if (isLedger(wallet)) return; // FIXME: Expected failure
+      if (!wallet || portis.isPortis(wallet)) return;
+      if (ledger.isLedger(wallet)) return; // FIXME: Expected failure
       if (!wallet.btcSupportsCoin("Testnet")) return;
-      let inputs = [
+      const tx: core.BitcoinTx = {
+        version: core.untouchable("tx.version not provided by test"),
+        locktime: core.untouchable("tx.locktime not provided by test"),
+        vin: core.untouchable("tx.vin not provided by test"),
+        vout: [
+          {
+            value: core.untouchable("tx.vout[0].value not provided by test"),
+            scriptPubKey: {
+              hex: "0014b31dc2a236505a6cb9201fa0411ca38a254a7bf1",
+            },
+          },
+        ],
+      };
+      const inputs: core.BTCSignTxInputUnguarded[] = [
         {
-          addressNList: bip32ToAddressNList("m/84'/1'/0'/0/0"),
-          scriptType: BTCInputScriptType.SpendWitness,
+          addressNList: core.bip32ToAddressNList("m/84'/1'/0'/0/0"),
+          scriptType: core.BTCInputScriptType.SpendWitness,
           amount: String(100000),
           vout: 0,
           txid: "e4b5b24159856ea18ab5819832da3b4a6330f9c3c0a46d96674e632df504b56b",
-          tx: {
-            vout: [
-              {
-                scriptPubKey: {
-                  hex: "0014b31dc2a236505a6cb9201fa0411ca38a254a7bf1",
-                },
-              },
-            ],
-          },
+          tx,
+          hex: undefined as string,
```

Comment:
We can't fill in `hex` from `tx` through the power of manual calculation because `tx` is missing a bunch of fields. (We should probably replace this test with something from the actual blockchain so we don't have to pull these shenanigans.)